Backend mockup

NolKeereweer opened the pull request:
A starting point for the backend.

NolKeereweer committed:
Setup for the backend

Created a basic security config that basically turns off all security.

Created Posts with all the basic necessities it needs, same with Users.
Also gave them a seeder.

Created application properties.

NolKeereweer committed:
Merge branch 'main' into Backend-mockup

mio-xenoblade commented on backend/src/main/java/nl/itvitae/twitbook/post/PostRepository.java:
```diff
@@ -0,0 +1,8 @@
+package nl.itvitae.twitbook.post;
+
+import java.util.UUID;
```
useless import?

mio-xenoblade commented on backend/src/main/java/nl/itvitae/twitbook/post/PostController.java:
```diff
@@ -0,0 +1,59 @@
+@RequestMapping("api/v1/posts")
+public class PostController {
+
+  @Autowired
```
private final ... with allargsconstructor is preferred over autowired

mio-xenoblade commented on backend/src/main/java/nl/itvitae/twitbook/post/PostController.java:
```diff
@@ -0,0 +1,59 @@
+    return postRepository.findById(id);
+  }
+
+  @PostMapping
```
different path would be preferred ("/new")

mio-xenoblade commented on backend/src/main/java/nl/itvitae/twitbook/user/UserController.java:
```diff
@@ -0,0 +1,22 @@
+public class UserController {
+
+  @Autowired
+  private UserRepository userRepository;
```
private final with all args constructor again

mio-xenoblade commented on backend/src/main/resources/application.properties:
```diff
@@ -1 +1,9 @@
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
+spring.jpa.hibernate.ddl-auto=create-drop
 
+app.jwt.secret=jf385gb3djfg42895hgwhjf824nf
```
secret is likely too short, not an issue now tho

NolKeereweer committed:
post-review

Changed the @Autowired usages to @AllArgs+private final usages.

Removed unused imports, whitespaced in between different import categories.

NolKeereweer committed:
Using actual date/time now

Had no setter for date/time in the constructors so it would always be null.
